lrauv_ignition_plugins/src/ScienceSensorsSystem.cc

```diff
@@ -635,161 +645,161 @@ void ScienceSensorsSystem::PostUpdate(const ignition::gazebo::UpdateInfo &_info,
   IGN_PROFILE_THREAD_NAME("ScienceSensorsSystem PostUpdate");
   IGN_PROFILE("ScienceSensorsSystem::PostUpdate");
 
-  // Only update and publish if data has been loaded and simulation is not
-  // paused.
-  if (this->dataPtr->initialized && !_info.paused)
-  {
-    double simTimeSeconds = std::chrono::duration_cast<std::chrono::seconds>(
-      _info.simTime).count();
+  this->RemoveSensorEntities(_ecm);
+
+  if (_info.paused)
+    return;
 
-    // Update time index
-    if (this->dataPtr->multipleTimeSlices)
+  // Delay reading data and generating octrees until spherical coordinates are
+  // received.
+  if (!this->dataPtr->sphericalCoordinatesInitialized)
+  {
+    if (this->dataPtr->world.SphericalCoordinates(_ecm))
     {
-      // Only update if sim time exceeds the elapsed timestamp in data
-      if (!this->dataPtr->timestamps.empty() &&
-        simTimeSeconds >= this->dataPtr->timestamps[this->dataPtr->timeIdx])
-      {
-        // Increment for next point in time
-        this->dataPtr->timeIdx++;
+      this->dataPtr->sphericalCoordinatesInitialized = true;
 
-        // Publish science data at the next timestamp
-        this->dataPtr->PublishData();
-      }
+      this->dataPtr->ReadData(_ecm);
+      this->dataPtr->GenerateOctrees();
     }
+  }
 
-    // Publish every n iters so that VisualizePointCloud plugin gets it.
-    // Otherwise the initial publication in Configure() is not enough.
-    if (this->dataPtr->repeatPubTimes % 10000 == 0)
+  if (!this->dataPtr->sphericalCoordinatesInitialized)
```

Comment:
Is this intended to be the same condition as the one above, or is this like an else inside the previous if-statement?

Reply:
It's like an `else`, moved in 193447d